library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,943 @@
+    @property
+    def description(self):
+        if self._values['description'] is None:
+            return None
+        elif self._values['description'] in ['none', '']:
+            return ''
+        return self._values['description']
+
+    @property
+    def translate_address(self):
+        if self._values['translate_address'] is None:
+            return None
+        if self._values['translate_address']:
+            return 'enabled'
+        else:
+            return 'disabled'
+
+    @property
+    def translate_port(self):
+        if self._values['translate_port'] is None:
+            return None
+        if self._values['translate_port']:
+            return 'enabled'
+        else:
+            return 'disabled'
+
+    @property
+    def advertise(self):
+        if self._values['advertise'] is None:
+            return None
+        return flatten_boolean(self._values['advertise'])
+
+    @property
+    def vlans_enabled(self):
+        if self._values['enabled_vlans'] is None:
+            return None
+        elif self._values['vlans_enabled'] is False:
+            # This is a special case for 'all' enabled VLANs
+            return False
+        if self._values['disabled_vlans'] is None:
+            return True
+        return False
+
+    @property
+    def vlans_disabled(self):
+        if self._values['disabled_vlans'] is None:
+            return None
+        elif self._values['vlans_disabled'] is True:
+            # This is a special case for 'all' enabled VLANs
+            return True
+        elif self._values['enabled_vlans'] is None:
+            return True
+        return False
+
+    @property
+    def enabled_vlans(self):
+        if self._values['enabled_vlans'] is None:
+            return None
+        elif any(x.lower() for x in self._values['enabled_vlans'] if x.lower() in ['all', '*']):
+            result = [fq_name(self.partition, 'all')]
+            return result
+        results = list(set([fq_name(self.partition, x) for x in self._values['enabled_vlans']]))
+        results.sort()
+        return results
+
+    @property
+    def disabled_vlans(self):
+        if self._values['disabled_vlans'] is None:
+            return None
+        elif any(x.lower() for x in self._values['disabled_vlans'] if x.lower() in ['all', '*']):
+            raise F5ModuleError(
+                'You cannot disable all VLANs. You must name them individually.'
+            )
+        results = list(set([fq_name(self.partition, x) for x in self._values['disabled_vlans']]))
+        results.sort()
+        return results
+
+    @property
+    def vlans(self):
+        disabled = self.disabled_vlans
+        if disabled:
+            return self.disabled_vlans
+        return self.enabled_vlans
+
+    @property
+    def irules(self):
+        results = []
+        if self._values['irules'] is None:
+            return None
+        if is_empty_list(self._values['irules']):
+            return ''
```

Comment:
return empty list []

Reply:
This comment is addressed now.

Reply:
return empty list

Reply:
I have the change. This comment is addressed now.